dbldatagen/spark_singleton.py

```diff
@@ -28,17 +27,25 @@ def getInstance(cls):
         return SparkSession.builder.getOrCreate()
 
     @classmethod
-    def getLocalInstance(cls, appName="new Spark session"):
+    def getLocalInstance(cls, appName="new Spark session", useAllCores=True):
```

Comment:
If we use this code only for tests, then I would suggest to use `pytest-spark` package instead because it allows to configure `SparkSession` via `pytest.ini` file

Reply:
Generally my preference is to minimize the number of additional packages being used that are not part of Databricks runtime or part of standard library

Over time, plan is to switch all tests to PyTest but it will be over multiple PRs

Reply:
It’s a dev dependency, not runtime dependency. I don’t see problem with that

Reply:
I responded directly but in any event the pytest-spark code doesnt auto detect the number of cores. Key goal of this PR is to improve the performance of unit test run